spec/controllers/admin/videos_controller_spec.rb

```diff
@@ -0,0 +1,93 @@
+require 'spec_helper'
+
+describe Admin::VideosController do
+  describe "GET new" do
+    it_behaves_like "requires sign in" do
+      let(:action) { get :new }
+    end
+
+    it_behaves_like "requires admin" do
+      let(:action) { get :new }
+    end
+
+    it "sets @video to new video" do
+      bob = Fabricate(:admin)
+      set_current_user(bob)
+      get :new
+      expect(assigns(:video)).to be_instance_of(Video)
+    end
+  end
+
+  describe "POST create" do
+    it_behaves_like "requires sign in" do
+      let(:action) { post :create }
+    end
+
+    it_behaves_like "requires admin" do
+      let(:action) { post :create }
+    end
+
+    context "with valid input" do
+      it "redirects to add new video page" do
+        bob = Fabricate(:admin)
+        set_current_user(bob)
+        category = Fabricate(:category)
+        video_attributes = Fabricate.attributes_for(:video)
+        post :create, video: video_attributes.merge!(category_id: category.id)
+        expect(response).to redirect_to new_admin_video_path
+      end
+
+      it "creates video" do
+        bob = Fabricate(:admin)
+        set_current_user(bob)
+        category = Fabricate(:category)
+        video_attributes = Fabricate.attributes_for(:video)
+        post :create, video: video_attributes.merge!(category_id: category.id)
+        expect(category.videos.count).to eq(1)
+      end
+
+      it "flashes success message" do
+        bob = Fabricate(:admin)
+        set_current_user(bob)
+        category = Fabricate(:category)
+        video_attributes = Fabricate.attributes_for(:video)
+        post :create, video: video_attributes.merge!(category_id: category.id)
```

Comment:
I think you don't need to merge in category, you can Fabricate it in step before along with video.